core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -31,4 +33,24 @@ private CatalogProperties() {
   public static final String HIVE_URI = "uri";
   public static final String HIVE_CLIENT_POOL_SIZE = "clients";
   public static final int HIVE_CLIENT_POOL_SIZE_DEFAULT = 2;
+
+  public static final String LOCK_IMPL = "lock.impl";
+
+  public static final String LOCK_HEARTBEAT_INTERVAL_MS = "lock.heartbeat-interval-ms";
+  public static final long LOCK_HEARTBEAT_INTERVAL_MS_DEFAULT = TimeUnit.SECONDS.toMillis(3);
+
+  public static final String LOCK_HEARTBEAT_TIMEOUT_MS = "lock.heartbeat-timeout-ms";
+  public static final long LOCK_HEARTBEAT_TIMEOUT_MS_DEFAULT = TimeUnit.SECONDS.toMillis(15);
+
+  public static final String LOCK_HEARTBEAT_THREADS = "lock.heartbeat-threads";
+  public static final int LOCK_HEARTBEAT_THREADS_DEFAULT = 4;
```

Comment:
Why would a lock need more than one heartbeat thread?

Reply:
Because all heartbeats are executed by a shared `ScheduledExecutorService`. If there are multiple commits ongoing for multiple tables, it will need more than 1 thread.